### ISSUE TITLE
API for unloading tabs from other addons doesn't work for new tab pages

### ISSUE THREAD

**Issue by Lej77** [rNeomy/auto-tab-discard#318]
## Description

My [Tab Unloader for Tree Style Tab](https://github.com/Lej77/tab-unloader-for-tree-style-tab) extension allows unloading tabs by sending a [`discard` message to Auto Tab Discard](https://github.com/rNeomy/auto-tab-discard/pull/50#issuecomment-391238140). This has worked really well but I noticed that recently this stopped working for special Firefox tabs such as new tab pages (`about:newtab`).

I debugged my extension and found that the Auto Tab Discard message was never responded to and therefore the message promise never resolved. I then debugged Auto Tab Discard itself and found that at [line 13 of `auto-tab-discard/core.mjs`](https://github.com/rNeomy/auto-tab-discard/blob/eb5e664a5f1f6eebafc1034fc33903b7f3e6d055/v3/worker/core.mjs#L13) the message handler throws an error because the `url` property is `undefined`. 

I think this issue should be fixed if a check is added so that `url` is "truthy", for example by changing the condition to `Boolean(url) && (url.startsWith('http') || url.startsWith('ftp')) && !discarded && !active`.

I thought about working around this in my extension but it doesn't have permission to access URLs so I can't check what tab is being unloaded before sending the message to Auto Tab Discard.

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install [Auto Tab Discard](https://addons.mozilla.org/firefox/addon/auto-tab-discard/).
 3. Install [Tab Unloader for Tree Style Tab](https://addons.mozilla.org/firefox/addon/tab-unload-for-tree-style-tab/).
 4. Go to `Tab Unloader for Tree Style Tab`'s settings page and in the `Other Settings` section enable the `Unload tabs via the Auto Tab Discard extension` option.
 5. Use `Tab Unloader for Tree Style Tab` to unload a "New Tab" page, for example by right clicking on such a tab and selecting the `Unload Tab(s)` context menu item.
    - If [Tree Style Tab](https://addons.mozilla.org/firefox/addon/tree-style-tab/) is installed then it is easier to see what tabs are actually unloaded since they are dimmed in the sidebar. (At least if `Tab Unloader for Tree Style Tab`'s style rules are registered, so might need to disable and re-enable `Tab Unloader for Tree Style Tab` after installing `Tree Style Tab`.)

## Expected result
The "New Tab" page should be unloaded.

## Actual result
The "New Tab" remains loaded.

## Environment

 * Platform (OS): Windows 10
 * Version of Firefox: 106.0.1
 * Version of Auto Tab Discard: 0.6.4
 * Version of Tab Unloader for Tree Style Tab: 6.15

**Comment by rNeomy** [comment]
Thanks for the report. Does [334d95b](https://github.com/rNeomy/auto-tab-discard/commit/334d95b6753480f9c8f5cb18aa5518c8a0f91227) fix the issue? btw, since ATD filters tabs from external requests (HTTP && HTTPS only), I am not sure how the blank one got discarded.

**Comment by rNeomy** [comment]
[ATD.zip](https://github.com/rNeomy/auto-tab-discard/files/9844260/ATD.zip)

**Comment by Lej77** [comment]
Thanks for the quick follow-up! I tested with `ATD.zip` and this issue seems to have been fixed!

The unloading of blank tabs is actually handled by my extension. It has some logic to handle tabs that cannot be unloaded by Auto Tab Discard at [line 139 of tab-unloader-for-tree-style-tab/background.js](https://github.com/Lej77/tab-unloader-for-tree-style-tab/blob/3d0ed09fc66f397494bad7619618dcf61cc05ce3/background/background.js#L139-L151). This affects tabs in private windows (if ATD hasn't been granted permission for private windows) and tabs that have weird URLs such as `about:newtab`. My extension's fallback code waits for ATD to finish with the unload request and since that didn't happen the fallback code never unloaded the blank tabs.
